src/Compilers/CSharp/Portable/Lowering/Instrumentation/ModuleCancellationInstrumenter.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.Shared.Collections;
+
+namespace Microsoft.CodeAnalysis.CSharp;
+
+/// <summary>
+/// Implements instrumentation for <see cref="CodeAnalysis.Emit.InstrumentationKind.ModuleCancellation"/>.
+/// </summary>
+internal sealed class ModuleCancellationInstrumenter(
+        MethodSymbol throwMethod,
+        SyntheticBoundNodeFactory factory,
+        Instrumenter previous)
+        : CompoundInstrumenter(previous)
+{
+    protected override CompoundInstrumenter WithPreviousImpl(Instrumenter previous)
+        => new ModuleCancellationInstrumenter(
+            throwMethod,
```

Comment:
>throwMethod [](http://example.com/codeflow?start=12&length=11)

We do not allow implicit capturing of primary constructor parameters under Compilers.